src/maps/OverlayedMap.cpp

```diff
@@ -146,6 +156,30 @@ void OverlayedMap::doWork() {
 }
 
 void OverlayedMap::setOverlayConfig(const OverlayConfig& conf) {
+    if (overlayConfig.drawAircraft != conf.drawAircraft) {

```

Comment:
I think this function would be more readable if it simply saved all settings, even if they didn't change. Since this is only triggered by a user action, I think the performance loss doesn't matter and should be negligible anyways.